hack/generated/controllers/controller_test.go

```diff
@@ -0,0 +1,113 @@
+*/
+
+package controllers_test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	storage "github.com/Azure/k8s-infra/hack/generated/apis/microsoft.storage/v20190401"
+	"github.com/Azure/k8s-infra/hack/generated/pkg/armclient"
+	"github.com/Azure/k8s-infra/hack/generated/pkg/genruntime"
+)
+
+func Test_ResourceGroup_CRUD(t *testing.T) {
+	// TODO: Can we put this someplace common?
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	g := NewGomegaWithT(t)
+	ctx := context.Background()
+
+	// Create a resource group
+	rg := testContext.NewTestResourceGroup()
+	err := testContext.KubeClient.Create(ctx, rg)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// It should be created in Kubernetes
+	g.Eventually(testContext.Ensure.ProvisioningComplete(ctx, rg)).Should(BeTrue())
+	g.Expect(testContext.Ensure.ProvisioningSuccessful(ctx, rg)).Should(BeTrue())
+
+	g.Expect(rg.Status.Location).To(Equal(testContext.AzureRegion))
+	g.Expect(rg.Status.Properties.ProvisioningState).To(Equal(string(armclient.SucceededProvisioningState)))
+	g.Expect(rg.Status.ID).ToNot(BeNil())
+	armId := rg.Status.ID
+
+	// Delete the resource group
+	err = testContext.KubeClient.Delete(ctx, rg)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Eventually(testContext.Ensure.Deleted(ctx, rg)).Should(BeTrue())
+
+	// Ensure that the resource group was really deleted in Azure
+	// TODO: Do we want to just use an SDK here? This process is quite icky as is...
+	exists, err := testContext.AzureClient.HeadResource(
+		ctx,
+		armId,
+		"2020-06-01")
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(exists).To(BeFalse())
+}
+
+func Test_StorageAccount_CRUD(t *testing.T) {
+	// TODO: Can we put this someplace common?
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	g := NewGomegaWithT(t)
+	ctx := context.Background()
+
+	// Custom namer because storage accounts have strict names
+	namer := testContext.Namer.WithSeparator("")
+
+	// Create a storage account
+	accessTier := storage.StorageAccountPropertiesCreateParametersAccessTierHot
+	acct := &storage.StorageAccount{
+		ObjectMeta: testContext.MakeObjectMetaWithName(namer.GenerateName("stor")),
+		Spec: storage.StorageAccounts_Spec{
+			Location:   testContext.AzureRegion,
+			ApiVersion: "2019-04-01", // TODO: This should be removed from the storage type eventually
+			Owner:      genruntime.KnownResourceReference{Name: sharedResourceGroup.Name},
+			Kind:       storage.StorageAccountsSpecKindBlobStorage,
+			Sku: storage.Sku{
+				Name: storage.SkuNameStandardLRS,
+			},
+			// TODO: They mark this property as optional but actually it is required
+			Properties: &storage.StorageAccountPropertiesCreateParameters{
+				AccessTier: &accessTier,
+			},
+		},
+	}
+	err := testContext.KubeClient.Create(ctx, acct)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// It should be created in Kubernetes
+	g.Eventually(testContext.Ensure.ProvisioningComplete(ctx, acct)).Should(BeTrue())
```

Comment:
Ensure returning a boolean means the error messages when this fails are not that useful -- maybe an `err` instead?

Reply:
Tried err and it was pretty ugly. Will attempt what @porges suggested of:
```
g.Eventually(acct).Should(BeProvisioned())
g.Expect(acct).Should(BeSuccessfullyProvisioned())
```

Reply:
Take a look at the new way -- I think it's quite a bit cleaner